src/allmydata/client.py

```diff
@@ -211,12 +212,12 @@ def init_introducer_clients(self):
         self.introducer_clients = []
         for introducer_furl in self.introducer_furls:
             tubID, location_hints, name = decode_furl(introducer_furl)
-            cfg = os.path.join(self.basedir, "private", "%s.introduced.yaml" % name)
+            config_filepath = FilePath(os.path.join(self.basedir, "private", "%s.introduced.yaml" % name))
```

Comment:
If ``name`` contained ``/`` or ``..``, then the FilePath wouldn't necessarily point below basedir/private. Use ``FilePath(self.basedir).child("private").child("%s.introduced.yaml" % (name,))``